website/files/models/onedrive.py

```diff
@@ -0,0 +1,16 @@
+from website.files.models.base import File, Folder, FileNode
+
+
+__all__ = ('OneDriveFile', 'OneDriveFolder', 'OneDriveFileNode')
+
+
+class OneDriveFileNode(FileNode):
```

Comment:
Assuming that OneDrive is fairly typical in how their files work (and from a quick glance over their API, that seems to be a valid assumption), these should be fine.